packages/components/MIGRATING.md

```diff
@@ -434,6 +437,76 @@ Changes:
 </SegmentedControl.Root>
 ```
 
+## Switch
+
+Changes:
+
+-   **States**:
+
+    -   `value` (controlled) and `defaultValue` (uncontrolled) replaces `mode`.
+    -   `indeterminate` state support has been dropped.
+    -   `required` is now available.
+    -   `name` is now available.
+
+-   **Accessibility**:
+
+    -   `aria-label` and `aria-labelledby` are both optional but one must be used.
+    -   `aria-describedby` has been added.
+
+-   **Event Handling**:
+
+    -   The `onChange` event now returns the latest state of the value rather than a `MouseEvent`.
+    -   `onBlur` and `onFocus` callbacks have been added.
+
+-   **Removed Properties**:
+
+    -   `label`, `labelStyle` and `tooltip` have been removed in favour of composition.
+
+#### Old
+
+```tsx
+<Switch
+    id="old-switch"
+    label="Dark mode"
+    tooltip="Toggle dark mode"
+    mode={true}
+    size="medium"
+    hug={true}
+    disabled={false}
+    onChange={(event) => console.log('Value:', event.target.value}
+    aria-label="Adjust volume level"
+    data-test-id="switch-test"
+/>
+```
+
+#### New
+
+```tsx
+<Tooltip.Root enterDelay={200}>
+    <Tooltip.Trigger>
+        <Label id="dark-mode-label" htmlFor="new-switch">Dark mode</Label>
+    </Tooltip.Trigger>
+    <Tooltip.Content side="top" padding="spacious">
+        <p id="dark-mode-switch-description">Toggle dark mode.</p>
+    </Tooltip.Content>
+</Tooltip.Root>
```

Comment:
This part of the example is wrong `tooltip` added a question-mark-circle icon with the tooltip text
![Screenshot 2024-07-23 at 13 34 40](https://github.com/user-attachments/assets/ab03af8d-ef82-4938-b825-00363d485e48)